src/android/com/outsystems/firebase/analytics/validator/OSFANLEventItemsValidator.kt

```diff
@@ -0,0 +1,122 @@
+package com.outsystems.firebase.analytics.validator
+
+import android.os.Bundle
+import com.outsystems.firebase.analytics.model.OSFANLDefaultValues
+import com.outsystems.firebase.analytics.model.OSFANLError
+import com.outsystems.firebase.analytics.model.OSFANLInputDataFieldKey.CUSTOM_PARAMETERS
+import com.outsystems.firebase.analytics.model.OSFANLInputDataFieldKey.ITEMS
+import com.outsystems.firebase.analytics.model.OSFANLInputDataFieldKey.ITEM_ID
+import com.outsystems.firebase.analytics.model.OSFANLInputDataFieldKey.ITEM_NAME
+import com.outsystems.firebase.analytics.model.OSFANLInputDataFieldKey.KEY
+import com.outsystems.firebase.analytics.model.OSFANLInputDataFieldKey.VALUE
+import com.outsystems.firebase.analytics.model.putAny
+import org.json.JSONArray
+
+/**
+ * Responsible for validating the Event items
+ * @property minLimit the minimum expected number of items
+ */
+class OSFANLEventItemsValidator(
+    private var minLimit: OSFANLMinimumRequired = OSFANLMinimumRequired.NONE
+) {
+
+    /**
+     * Validates a JSONArray of items
+     * @param items the array of items to validate
+     * @return a list of bundles, each representing a validated item
+     */
+    fun validate(items: JSONArray): List<Bundle> {
+
+        var inputItems = items
+        val result = mutableListOf<Bundle>()
+
+        // validate minimum limit
+        when (minLimit) {
+            // no processing is needed
+            OSFANLMinimumRequired.NONE -> if(inputItems.length() == 0) return result
+            OSFANLMinimumRequired.AT_LEAST_ONE -> if (inputItems.length() == 0)
+                throw OSFANLError.missing(ITEMS.json)
+            OSFANLMinimumRequired.ONE -> if (inputItems.length() == 0)
+                throw OSFANLError.tooMany(ITEMS.json, OSFANLDefaultValues.itemQuantity)
```

Comment:
Is this error correct? I believe the issue is that the items are missing, not that there are too many items.

Also, the `if(inputItems.length() == 0)` is always performed, so maybe the `when` is misplaced.